src/components/EarnDashboard/components/ProviderDetails/OpportunityRow.tsx

```diff
@@ -126,34 +112,51 @@ export const OpportunityRow: React.FC<
     ))
   }, [apy, cryptoAmountBaseUnit, group, type])
 
-  const renderRewardAssets = useMemo(() => {
-    if (!nestedAssetIds) return null
+  const renderNestedAssets = useMemo(() => {
     return (
       <List style={{ marginTop: 0 }}>
-        {nestedAssetIds.map(assetId => {
-          if (bnOrZero(underlyingAssetBalances[assetId]?.cryptoBalancePrecision).eq(0)) return null
+        {Object.entries(rewardsBalances).map(([rewardAssetId, rewardBalance]) => {
+          if (bnOrZero(rewardBalance.cryptoBalancePrecision).isZero()) return null
           return (
             <NestedAsset
-              key={assetId}
+              key={rewardAssetId}
               isClaimableRewards={isClaimableRewards}
               isExternal={opportunity.isReadOnly}
-              assetId={assetId}
-              balances={underlyingAssetBalances[assetId]}
+              assetId={rewardAssetId}
+              balances={rewardBalance}
               onClick={() => handleClick(DefiAction.Claim)}
-              type={translate(underlyingAssetLabel)}
+              type={translate('common.reward')}
             />
           )
         })}
+        {Object.entries(underlyingAssetBalances).map(
+          ([underlyingAssetId, underlyingAssetBalance]) => {
+            // Don't display the same asset as an underlying, that's an implementation detail, but shouldn't be user-facing
+            if (underlyingAssetId === assetId) return null
+            if (bnOrZero(underlyingAssetBalance.cryptoBalancePrecision).isZero()) return null
+            return (
+              <NestedAsset
+                key={underlyingAssetId}
+                isClaimableRewards={isClaimableRewards}
+                assetId={underlyingAssetId}
+                balances={underlyingAssetBalance}
+                onClick={() => history.push(`/assets/${underlyingAssetId}`)}
```

Comment:
Just for the sake of making `onClick` happy (it is required) and to enable the click if we so wish in the future, at the moment, `<NestedAsset />` only makes rewards clickable, not underlying assets